### ISSUE TITLE
Custom orders and preset filters

### ISSUE THREAD

**Issue by justincy** [genealogysystems/fs-traversal#29]
Right now we have preset orders and custom filters. We should add support for custom orders and preset filters. This will allow us to ship and wider range of options while better decoupling the two concepts (they're a little mixed right now).

**Comment by justincy** [comment]
### Order

Right now `order` is a string. When the weight of a person is calculated, the string is used in a switch statement to find the right code or function that actually calculates the weight. 

Instead of storing the string we will store a reference to the function. We allow for presets (passing in a string) by mapping the string to a set of built-in functions.

The biggest issue is the fetched objects. In them we store the values of the WRD variables. It feels wrong to track WRD state in the fetched objects without allowing custom order functions to do the same. 

We could create a mechanism for allowing order functions to store data in the fetched state. This would likely require another optional function that gets called when the fetch objects are created and returns the necessary state values to be used later when calculating the weight. (These are two separate steps because the fetched objects are generated and passed into the filter function; weights are not calculated until after the filter functions).

Now that I think about it, there's no reason why we need to store this state in the fetched object. It can all be calculated by examining the path that is stored in the objects.

In conclusion:

* Get rid of order state being stored in fetched objects. Calculate on-demand based off of the path.
* Change order to store a reference to proper weight calculation function instead of doing a lookup each time.
* Allow for presets by passing a string (using the current values) and custom orders by passing in the weight calc function.

**Comment by justincy** [comment]
### Filter

Filter is much more simple. We just need to modify it to accept a string matching a set of built-in filter functions.
